### PR TITLE
[SPARK-48671][SQL][TESTS] Add test cases for `Hex.hex`

### PR DESCRIPTION
**Pull request by wayneguow**
### What changes were proposed in this pull request?

This PR aims  to add some test cases  for function `Hex.hex(bytes: Array[Byte])`.

### Why are the changes needed?

Supplement test cases to make the code more robust.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Passed new test cases added in `HexSuite`.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by wayneguow** [comment]
cc @yaooqinn  thanks.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala [hunk 1]
```diff
@@ -37,4 +37,24 @@ class HexSuite extends SparkFunSuite {
...
   }
+
+  test("SPARK-48644: hex bytes values") {
+    assert(Hex.hex(Array[Byte]()).toString === "")
+    assert(Hex.hex(Array[Byte](0, 1, 15, 16, 17, Byte.MaxValue)).toString === "00010F10117F")
+    assert(Hex.hex(Array[Byte](-1, -15, -16, -17, Byte.MinValue)).toString === "FFF1F0EF80")
+  }
+
+  test("SPARK-48644: bytes length check in hex") {
+    val bytes = new Array[Byte](1073741824)
```
Comment (yaooqinn) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala:48]: I didn't add this as it's quite memory consuming

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala [hunk 2]
```diff
@@ -37,4 +37,24 @@ class HexSuite extends SparkFunSuite {
     assert(Hex.hex(Long.MinValue).toString === "8000000000000000")
     assert(Hex.hex(Long.MaxValue).toString === "7FFFFFFFFFFFFFFF")
   }
+
+  test("SPARK-48644: hex bytes values") {
```
Comment (yaooqinn) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala:41]: This test case is irrelevant to SPARK-48644 and we do have test coverage for hex byte arrays

Comment (wayneguow) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala:41]: @yaooqinn How about submitting a new issue for keeping those test cases?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala (hunk 1 above)
Comment (wayneguow) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala:48]: Thank you for pointing out. Make sense.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala [hunk 3]
```diff
@@ -37,4 +37,10 @@ class HexSuite extends SparkFunSuite {
     assert(Hex.hex(Long.MinValue).toString === "8000000000000000")
     assert(Hex.hex(Long.MaxValue).toString === "7FFFFFFFFFFFFFFF")
   }
+
+  test("SPARK-48671: hex bytes values") {
+    assert(Hex.hex(Array[Byte]()).toString === "")
```
Comment (amaliujia) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala:42]: assert(Hex.hex(Array[Byte](null)).toString) === ??? 

is this a valid test case?

Comment (wayneguow) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/HexSuite.scala:42]: Yes. Both before and after `SPARK-48644` improvement, the result is "".

**Pull request closed without merging**